addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,241 @@
+                    INNER JOIN account_move move ON move.id = aml.move_id
+                    INNER JOIN res_company c ON aml.company_id = c.id
+                    INNER JOIN res_currency cur ON c.currency_id = cur.id
+                    LEFT JOIN account_report_expression expression ON expression.engine = 'tax_tags' AND expression.formula = SUBSTRING({acc_tag_name} from 2)
+                    LEFT JOIN account_report_line trl ON expression.report_line_id = trl.id
+                    WHERE aml.id IN %s
+                ),
+                tag_values AS (
+                    SELECT id,
+                        array_to_string(
+                            array_agg(name || ': ' || CASE WHEN cur_pos = 'before' THEN   currency || ' ' || tag_amount
+                                                           ELSE                         tag_amount || ' ' || currency
+                                                           END
+                            ),
+                            '        '
+                       ) AS tax_audit
+                    FROM tags
+                    GROUP BY id
+                )
+                UPDATE account_move_line
+                SET tax_audit = tag_values.tax_audit
+                FROM tag_values
+                WHERE tag_values.id = account_move_line.id
+            """
+            self.env.cr.execute(query, (tuple(ids),))
+        self.env.invalidate_all()
+
+    def _modify_tag_to_aml_relation(self, company_id, date_from):
+        """ Update Journal Items' tax grids to match current taxes' configuration.
+        The next query work in 3 steps.
+        1) Get a duo: (aml_id, tag_id or NULL) for each aml involved with the tax.
+            This first step is achieved in the 3 first table: base line, tax line, fusion.
+            a) As the base line and tax line aren't linked to tax in the same way, we need to gather them separately.
+            The base line get its repartition line by going through the tax.
+            b) The tax lines already have their repartition line linked.
+        2) Delete the previous relation aml - tag for each aml appearing in the previous queries.
+        3) Insert the new relation aml - tag.
+        4) Gather the ids of impacted amls into an array and return it.
+        :param: int company_id id of company
+        :return: list of impacted account.move.line ids
+        """
+        self.env.flush_all()
+        self.env.cr.execute("""
+            -- 1.a) Handle base line: relation aml <-> tag, if no relation, tag is NULL
+            WITH base_aml_id_rep_tag_to_insert AS (
+                SELECT aml.id AS aml_id, rep_tags.account_account_tag_id AS tag_id
+                FROM account_move_line aml
+                JOIN account_move move
+                ON aml.move_id = move.id AND move.company_id = %(company_id)s AND aml.date >= %(date_from)s
+                LEFT JOIN account_move caba_origin_move
```

Comment:
I see you joined the dark side of the force for the query formating :rofl: 
![image](https://github.com/user-attachments/assets/fa155882-21e8-416f-9590-b5936e59d709)

Reply:
I've been asked to :eyes: :innocent: